consensus/src/chained_bft/epoch_manager.rs

```diff
@@ -88,7 +108,101 @@ impl<T: Payload> EpochManager<T> {
     }
 
     pub fn epoch(&self) -> u64 {
-        self.epoch_info.read().unwrap().epoch
+        self.epoch_info.epoch
+    }
+
+    pub async fn check_epoch(
+        &mut self,
+        peer_id: &AccountAddress,
+        msg: FromNetworkMsg<T>,
+    ) -> anyhow::Result<EpochCheck<T>> {
+        match msg {
+            FromNetworkMsg::Proposal(proposal) => {
+                if proposal.epoch() != self.epoch_info.epoch {
```

Comment:
I see how it could make sense to "unify this repetitive code", and I don't see how it would somehow reduce the complexity of the rest of this PR. This code was previously in `NetworkTask`. 

The commit reducing the repetition could be completely separate from this PR, and done in a follow-up. How does doing it first make the overall restructuring easier?

Reply:
Ok I separated this epoch checking into a separate method. Every little bit helps!